api/client/client.go

```diff
@@ -234,22 +236,45 @@ func (c *Client) WriteTransaction(storeID, modelName string) (*WriteTransaction,
 	return &WriteTransaction{client: client, storeID: storeID, modelName: modelName}, nil
 }
 
-// Listen provides an update whenever the specified model is updated
-func (c *Client) Listen(storeID, modelName, entityID string) (<-chan ListenEvent, func()) {
+// Listen provides an update whenever the specified store, model type, or model instance is updated
+func (c *Client) Listen(storeID string, listenOptions ...es.ListenOption) (<-chan ListenEvent, func(), error) {
 	channel := make(chan ListenEvent)
 	ctx, cancel := context.WithCancel(c.ctx)
-	go func() {
-		defer close(channel)
-		req := &pb.ListenRequest{
-			StoreID:   storeID,
-			ModelName: modelName,
-			EntityID:  entityID,
+	filters := make([]*pb.ListenRequest_Filter, len(listenOptions))
+	for i, listenOption := range listenOptions {
+		var action pb.ListenRequest_Filter_Action
+		switch listenOption.Type {
+		case es.ListenAll:
+			action = pb.ListenRequest_Filter_ALL
+		case es.ListenCreate:
+			action = pb.ListenRequest_Filter_CREATE
+		case es.ListenDelete:
+			action = pb.ListenRequest_Filter_DELETE
+		case es.ListenSave:
+			action = pb.ListenRequest_Filter_SAVE
+		default:
+			cancel()
+			return nil, nil, fmt.Errorf("unknown ListenOption.Type %v", listenOption.Type)
 		}
-		stream, err := c.client.Listen(ctx, req)
-		if err != nil {
-			channel <- ListenEvent{err: err}
-			return
+		filters[i] = &pb.ListenRequest_Filter{
+			ModelName: listenOption.Model,
+			EntityID:  listenOption.ID.String(),
+			Action:    action,
 		}
+	}
+	req := &pb.ListenRequest{
+		StoreID: storeID,
+		Filters: filters,
+	}
+	stream, err := c.client.Listen(ctx, req)
+	if err != nil {
+		cancel()
+		return nil, nil, err
+	}
+	go func() {
+		defer close(channel)
+
+	L:
```

Comment:
Label here to break out of the switch nested inside the for loop.

Reply:
Maybe a more descriptive label? Just a thought

Reply:
Cool, done